### DISCUSSION TITLE
Confusion with Proxies and EIP-1667

### DISCUSSION THREAD

**Discussion by Ifra001** [Cyfrin/foundry-full-course-cu#2953] [General]
I am in the upgradable smart contract section of the course and it is a little bit confusing. The assembly stuff is hard to understand and overall my brain is just fried

Please give me some resources and articles to learn more about proxies and how they interact with the implementation contract. I understand the logic but I don't understand it when it is implemented in code. The code bit is confusing

**Comment by EngrPips** [comment]
Here https://www.cyfrin.io/blog/upgradeable-proxy-smart-contract-pattern is a Cyfrin post that explains Upgradeable smart contracts. Check it out.
